extension/app/ts/background/background.ts

```diff
@@ -255,7 +256,7 @@ export async function appendTransactionToSimulator(transaction: EthereumUnsigned
 	if ( simulator === undefined) return
```

Comment:
what do you mean by missed opportunity?

Reply:
moving the anonymous async function passed to  `updateSimulationState` in #L257 on a separate execution line (up one line) would already ensure `simulation` object is not undefined so there would be no need to use optional chaining `?.`. it seems you needed it here because it is out of context for typescript?

Reply:
ah, its asynchronous code. In your modified version, you would end up with wrong state if new transaction is appended to the simulation between the awaits, as then the code would return old state

Reply:
Your proposed change would alter *when* the `simulationWithTransaction` occurs, which may be undesirable.  If it is acceptable to append the transaction sooner rather than later, then I think refactoring as proposed would be a good idea.